pkg/imgpkg/bundle/bundle.go

```diff
@@ -142,12 +158,21 @@ func (o *Bundle) pull(baseOutputPath string, ui goui.UI, pullNestedBundles bool,
 		return err
 	}
 
-	localizedImagesLockToRepo, notLocalizedToBundle, err := NewImagesLock(imagesLock, o.imgRetriever, o.Repo()).LocalizeImagesLock()
+	lock := NewImagesLock(imagesLock, o.imgRetriever, o.Repo(), locationFetcher)
+	localizedImagesLockToRepo, notLocalizedToBundle, err := lock.LocalizeImagesLock()
 	if err != nil {
 		return err
 	}
 
 	if pullNestedBundles {
+		imageBundles := map[string]bool{}
+		fetch, err := locationFetcher.Fetch()
```

Comment:
renamed to `locationsConfig`

> even if we have this information cached, don't you feel like it is redundant to call the fetch here again since we have already done that on LocalizeImagesLock?

Will address this in a follow up refactor. With the goal of removing ImagesLock.Localized and instead getting ImageRefs with Bundle information (removing the need of calling Fetch again)